[VIRTS-2046] Filter GUI Operations

emmanvg opened the pull request:
## Description

This PR adds a new search box and criteria for filtering data out of big operations. The only missing requirement from this work is filtering by output, since this data is loaded on demand (user clicks View Output) it may pose a problem if we need to pre-load all that data at once affecting responsiveness. Observe the image below for user rendering.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tested using the Mock plugin (has a bug on the latest Caldera) using the Thief adversary to generate dummy data. Select your search criteria and type down your search and the data will get filtered down.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

![image](https://user-images.githubusercontent.com/8574582/154503600-0242f84d-a57c-4a55-a178-6cb597371c74.png)

argaudreau commented on templates/operations.html:
```diff
@@ -178,6 +178,31 @@ <h3 x-text="selectedOperation.name"></h3>
+                            <div class="mb-3">
+                                <input x-model="searchText" placeholder="Search..." type="search" class="is-small">
+                            </div>
+                            <div>
```
It'd also be useful to have a "Clear" button that will delete the filter term, and you can set it to only be visible when there's a filter term.

emmanvg replied:
Moved the filter up along Manual Command and Potential Link. Added a Clear button that clears both the text and search criteria.

argaudreau commented on templates/operations.html:
```diff
@@ -1114,6 +1139,40 @@ <h3 x-text="selectedOperation.name"></h3>
+            searchText: '',
+            searchCriteria: 'decide',
+
+            get filterFromSelectedOperation() {
```
Super nit-picky, but I'd rename this to `filterLinksFromSelectedOperation` only because there are a ton of functions on this page so it helps to be explicit.

emmanvg replied:
Done, and also changed variable names.

emmanvg commented:
@argaudreau thanks for the feedback! I implemented all of the changes you proposed. See the new UI images below, but summarizing:

- Added the style class to <input> element.
- Changed <input> placeholder to use filter instead of search. Likewise updated the JS variables to reflect that change, they are now `filterText` and `filterCriteria`.
- With these changes to the dropdown plus the other classes, you can delete a filter text (look at tiny X next to the term) or use the Clear button (which removes the text and filter criteria)

![image](https://user-images.githubusercontent.com/8574582/157766180-42434e2c-f95e-4779-8399-5edc53fc9ce4.png)
![image](https://user-images.githubusercontent.com/8574582/157766218-d5aa4193-b6de-478f-8fb2-7d6fcd9f6f0f.png)

emmanvg commented on templates/operations.html:
```diff
@@ -178,6 +178,31 @@ <h3 x-text="selectedOperation.name"></h3>
                     <div class="timeline-table">
+                        <div class="is-flex is-flex-direction-row is-justify-content-space-between">
+                            <div class="mb-3">
+                                <input x-model="searchText" placeholder="Search..." type="search" class="is-small">
```
Both made. 👍

cmagone reviewed (dismissed):
This is a good start on this functionality. Next, we should add the ability to declare multiple values. So if I want to see paw#1 and paw#3, I can specify both values.
